lib/elixir/lib/map_set.ex

```diff
@@ -409,21 +332,9 @@ defmodule MapSet do
   """
   @doc since: "1.14.0"
   @spec filter(t(a), (a -> as_boolean(term))) :: t(a) when a: value
-  def filter(%MapSet{map: map}, fun) when is_map(map) and is_function(fun) do
-    iter = :maps.iterator(map)
-    next = :maps.next(iter)
-    keys = filter_keys(next, fun)
-    %MapSet{map: Map.from_keys(keys, @dummy_value)}
-  end
-
-  defp filter_keys(:none, _fun), do: []
-
-  defp filter_keys({key, _value, iter}, fun) do
-    if fun.(key) do
-      [key | filter_keys(:maps.next(iter), fun)]
-    else
-      filter_keys(:maps.next(iter), fun)
-    end
+  def filter(%MapSet{set: set} = map_set, fun) when is_function(fun) do
+    pred = fn element -> !!fun.(element) end
+    %{map_set | set: :sets.filter(pred, set)}
```

Comment:
Is it ok to use `:sets.filter/2` with an on-the-fly predicate, or should also go with `:sets.fold/3` here?

Reply:
I would also keep it as it was, even though it is opaque.

Reply:
I think the new version is fine. :)